platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java

```diff
@@ -385,6 +387,7 @@ public class SwirldsPlatform implements Platform, Startable {
                 wiring.wireAppCommunicationComponent(notificationEngine);
 
         preconsensusEventFileManager = buildPreconsensusEventFileManager(emergencyRecoveryManager);
+        clearPCESOnSoftwareUpgradeIfConfigured(softwareUpgrade, preconsensusEventFileManager);
```

Comment:
Nitpick, perhaps it would be nice to invoke this method inside `buildPreconsensusEventFileManager()`. The constructor is SUPER long right now, and over the long term I'm hoping if we can hide details inside helper methods.

Reply:
I missed this comment, I can move the functionality to the constructor.  That's why I made a method for it to begin with.  Wouldn't moving the task to the constructor violate the directive that the FileManager shouldn't know about software upgrades?

Reply:
I wouldn't move it to the constructor, I'd just move it into the method `buildPreconsensusEventFileManager()`, which is defined inside `SwirldsPlatform.java`.